File: November/4/transposefile.cpp
// LeetCode 194: Transpose File (C++ Equivalent)
// Explanation:
// 1. Read file lines, split by space into matrix.
// 2. Transpose matrix and print.
// Time Complexity: O(n * m)
// Space Complexity: O(n * m)

#include <iostream>
#include <sstream>
#include <vector>
#include <string>
using namespace std;

int main() {
    vector<vector<string>> rows;
    string line;
    while (getline(cin, line)) {
        istringstream iss(line);
        vector<string> row;
        string word;
        while (iss >> word) row.push_back(word);
        rows.push_back(row);
    }

    int cols = rows[0].size();
    for (int i = 0; i < cols; ++i) {
        for (int j = 0; j < (int)rows.size(); ++j) {
            if (j > 0) cout << " ";
            cout << rows[j][i];
        }
        cout << endl;
    }
    return 0;
}

File: November/3/transposefile.py
# LeetCode 194: Transpose File (Python Equivalent)
# Explanation:
# 1. Read all lines and split by whitespace.
# 2. Use zip(*rows) to transpose the 2D list.
# Time Complexity: O(n * m)
# Space Complexity: O(n * m)

class Solution:
    def transposeFile(self, data: str):
        lines = [line.split() for line in data.strip().split('\n')]
        transposed = zip(*lines)
        return [' '.join(row) for row in transposed]

sol = Solution()
data = "name age\nalice 21\nryan 30"
print('\n'.join(sol.transposeFile(data)))
# Output:
# name alice ryan
# age 21 30
